pkg/controller/daemon/daemoncontroller.go

```diff
@@ -553,6 +553,7 @@ func storeDaemonSetStatus(dsClient unversionedextensions.DaemonSetInterface, ds
 
 	var updateErr, getErr error
 	for i := 0; i < StatusUpdateRetries; i++ {
+		toUpdate.Status.ObservedGeneration = ds.Generation
```

Comment:
You mean to run update only when ds.Status.ObservedGeneration is smaller than ds.Generation?
Something like:

```go
if ds.Status.ObservedGeneration >= ds.Generation {
  return nil
}